docs/source/library-user-guide/api-health.md

```diff
@@ -69,3 +69,32 @@ For example:
 Deprecated methods will remain in the codebase for a period of 6 major versions or 6 months, whichever is longer, to provide users ample time to transition away from them.
 
 Please refer to [DataFusion releases](https://crates.io/crates/datafusion/versions) to plan ahead API migration
+
+## Migration Guidelines
+
+To ensure smooth upgrades and maintain application stability, the following guidelines must be followed for changes involving:
```

Comment:
Maybe we can also explicitly say here "it is preferrable not to make breaking API changes and instead add new APis and deprecate existing ones in which case no migration guide is needed"

That would perhaps also subtley incentivize people to avoid breaking changes 🤔

Reply:
I think we covered it in https://datafusion.apache.org/library-user-guide/api-health.html#breaking-changes ?